core/block_validator.go

```diff
@@ -80,30 +64,17 @@ func (v *BlockValidator) ValidateBlock(block *types.Block) error {
 	if _, err := state.New(parent.Root(), v.bc.chainDb); err != nil {
 		return ParentError(block.ParentHash())
 	}
-
+	// Header validity is known at this point, check the uncles and transactions
```

Comment:
When importing blocks or headers, previously we only checked PoW, and the block validator did everything else. As of this PR, the `consensus.Engine` checks both PoW as well as header consensus fields, since they are different depending on whether we use `ethash` or `clique`. Block contents are the same, so I left them unchanged.